### PR TITLE
Add cling variant for 0.9

### LINKED ISSUE #144: Towards packaging cling 0.9: cling variant of clangdev 9.x
Cling 0.9, which was released a few months ago requires clangdev 9.x, which the previous versions were using clangdev 5.x.

Unfortunately, both versions require a clangdev variant including some patches. There is a similar set of patches for the clangdev variant required for thr root project

For the [5.x version](https://github.com/conda-forge/clangdev-feedstock/tree/5.x/recipe), we were using the root-project git fork of clangdev to get a version of clang that was requiring these patches, while for 9.x, [the patches are included in the recipe](https://github.com/conda-forge/clangdev-feedstock/tree/9.x/recipe).

@chrisburr what do you think is the best approach here? Is there a reason why we now include the root patches instead of using their fork?

### PR DESCRIPTION
**Pull request by SylvainCorlay**
Fixes #144.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by SylvainCorlay** [comment]
May have to set -mcmodel=medium for ppc64.

**Comment by SylvainCorlay** [comment]
Before going with `-mcmodel=medium`, @serge-sans-paille suggested we try dropping `-fno-plt` on ppc.

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -21,3 +21,5 @@ variant:
   - default
   - hcc   # [linux]
   - root_62400  # [linux or osx]
+  - cling_v0.9  # Tag is cling-v0.8
+
```
Comment (h-vetinari) [inline recipe/conda_build_config.yaml:25]:
```suggestion
  - cling_v0.9  # Tag is cling-v0.9
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -133,6 +133,71 @@ source:
       - patches/root/0108-Mark-the-overridden-transient-file.patch                    # [variant == "root_62400"]
       - patches/root/0109-Add-missing-include-to-fix-Dcxxmodules-On.patch             # [variant == "root_62400"]
       - patches/root/0110-Import-various-changes-to-the-global-module-index.patch     # [variant == "root_62400"]
+
+      # Taken from cloning http://root.cern/git/clang.git and running
+      # git format-patch cling-v0.9..bed37d4b18ce6993fc2567e8de61f6b30fd0cb75
+      # Corresponds to Cling 0.0
```
Comment (h-vetinari) [inline recipe/meta.yaml:139]: Should this be 0.9 here instead of 0.0?

**Comment by SylvainCorlay** [comment]
Hum, I don't know what is going on with ppc64 and the root variant - especially since this error occurs without me changing anything to this variant.

I am eager to get his in as-is as this is blocking the cling 0.9 release.

Any thought @chrisburr?

**Comment by SylvainCorlay** [comment]
The reason may be related to the update to gcc 9.4.
The last successful build was using gcc 9.3.

**Comment by SylvainCorlay** [comment]
OK, the ppc build has been hanging for over 5 hours. I have no idea what to do.

**Pull request merged**